lib/src/Uncertainty/Algorithm/Sensitivity/HSICStatImplementation.cxx

```diff
@@ -0,0 +1,62 @@
+
+//                                               -*- C++ -*-
+/**
+ * @brief HSICStatImplementation implements the HSIC sensivity index for one marginal.
+ *
+ *  Copyright 2005-2021 Airbus-EDF-IMACS-ONERA-Phimeca
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#include "openturns/Pointer.hxx"
+#include "openturns/Exception.hxx"
+#include "openturns/Log.hxx"
+#include "openturns/HSICStatImplementation.hxx"
+#include "openturns/PersistentObjectFactory.hxx"
+BEGIN_NAMESPACE_OPENTURNS
+
+static const Factory<HSICStatImplementation> Factory_HSICStatImplementation;
+
+CLASSNAMEINIT(HSICStatImplementation)
+/** Default constructor */
+HSICStatImplementation::HSICStatImplementation(): PersistentObject()
+{
+  // Nothing
+};
+
+/** Virtual constructor */
+HSICStatImplementation * HSICStatImplementation::clone() const
+{
+  return new HSICStatImplementation(*this);
+}
+
+/** Compute the HSIC index for one marginal*/
+Scalar HSICStatImplementation::computeHSICIndex(const Sample&, const Sample&, const CovarianceModel&, const CovarianceModel&, const SquareMatrix&) const
+{
+  return 0.0 ;
```

Comment:
I disagree with this. The mother class should throw, not output arbitrary values.